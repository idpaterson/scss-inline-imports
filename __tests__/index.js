// Copyright (c) 2016-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

/* global it, expect */

const _ = require('lodash')
const path = require('path')

const inlineImports = require('../')

const fixture = path.resolve.bind(path, __dirname, 'fixtures')

const mapImports = imports =>
  imports
    .map(i => _.mapKeys(i, (v, k) => k.replace(__dirname, '')))
    .map(i => _.mapValues(i, mapImports))

const mapImportsFlattened = imports =>
  imports.map(i => i.replace(__dirname, ''))

it('inlines imports', () => {
  const { scss } = inlineImports(fixture('a.scss'))
  expect(scss).toMatchSnapshot()
})

it('inlines imports with includePaths', () => {
  const { scss } = inlineImports(fixture('b.scss'), {
    includePaths: [fixture('modules')]
  })
  expect(scss).toMatchSnapshot()
})

it('throws an error for missing imports', () => {
  expect(() => {
    inlineImports(fixture('b.scss'))
  }).toThrow(/not found/)
})

it('returns the imports', () => {
  const { imports } = inlineImports(fixture('a.scss'))
  expect(mapImports(imports)).toMatchSnapshot()
})

it('returns the importsFlattened', () => {
  const { importsFlattened } = inlineImports(fixture('a.scss'))
  expect(mapImportsFlattened(importsFlattened)).toMatchSnapshot()
})
