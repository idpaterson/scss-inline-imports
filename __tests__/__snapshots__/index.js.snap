// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inlines imports 1`] = `
".my-partial {}


.a {}

.world-c {}

.my-partial {}


.world-b {}

"
`;

exports[`inlines imports with includePaths 1`] = `
".my-partial {}

.my-module {}

"
`;

exports[`returns the imports 1`] = `
Array [
  Object {
    "/fixtures/a.scss": Array [
      Object {
        "/fixtures/_partial.scss": Array [],
      },
      Object {
        "/fixtures/hello/hello-b.scss": Array [
          Object {
            "/fixtures/hello/world/world-c.scss": Array [],
          },
          Object {
            "/fixtures/_partial.scss": Array [],
          },
        ],
      },
    ],
  },
]
`;

exports[`returns the importsFlattened 1`] = `
Array [
  "/fixtures/_partial.scss",
  "/fixtures/hello/world/world-c.scss",
  "/fixtures/_partial.scss",
  "/fixtures/hello/hello-b.scss",
  "/fixtures/a.scss",
]
`;
